libcxx/test/support/asan_testing.h

```diff
@@ -74,17 +74,11 @@ TEST_CONSTEXPR bool is_string_asan_correct(const std::basic_string<ChrT, TraitsT
   if (TEST_IS_CONSTANT_EVALUATED)
     return true;
 
-  if (!is_string_short(c) || _LIBCPP_SHORT_STRING_ANNOTATIONS_ALLOWED) {
```

Comment:
Do I understand correctly that this is where the largest part of test coverage comes from?

Reply:
I don't understand the question. After that patch, short and long strings should be annotated and it's tested here. Now we test if short strings are not poisoned. This change should increase strength of testing ASan.

Reply:
We have a bunch of existing tests where we use short strings and where you do `assert(is_string_asan_correct(s))` in the test suite. Before this patch, this would be a no-op whenever `s` was a short string, i.e. it wouldn't test anything.

My question / comment was asking for a confirmation that all methods like `basic_string::at`, `basic_string::resize` & friends are basically tested now due to this specific hunk, which makes the existing `assert`s in the test suite not no-ops anymore. If you said no, then I'd complain that we need to add tests for e.g. checking the asan-correctness of short strings w.r.t. `basic_string::resize`, and that for all other methods.

Reply:
Yes, now everything is tested. In that sense, it adds coverage. Yes.